File: 1. Java/day06/src/MinusAccount.java
/**
 * Account를 확장한 마이너스 계좌
 * @author hojin
 *
 */
public class MinusAccount extends Account{

	private long borrowMoney;

	public MinusAccount() {
		super();
	}

	public MinusAccount(String accountNum, String accountOwner) {
		super(accountNum, accountOwner);
	}

	public MinusAccount(String accountNum, String accountOwner, int passwd, long restMoney, long borrowMoney) {
		super(accountNum, accountOwner, passwd, restMoney);
		this.borrowMoney = borrowMoney;
	}
	
	public long getBorrowMoney() {
		return borrowMoney;
	}

	public void setBorrowMoney(long borrowMoney) {
		this.borrowMoney = borrowMoney;
	}

	@Override
	public long deposit(long money) {
		super.setRestMoney(super.getRestMoney()+money);
		return super.getRestMoney();
	}

	@Override
	public long getRestMoney() {
		return super.getRestMoney()-getBorrowMoney();
	}
	
	public static void main(String[] args) {
		MinusAccount minusAccount = new MinusAccount();
		System.out.println(minusAccount.getBorrowMoney());
		
		MinusAccount minusAccount2 = new MinusAccount("1231-1234-2222", "minus", 4321, 0, 1000000);
		minusAccount2.deposit(100000);
		System.out.println(minusAccount2.getRestMoney());
		
	}
}

File: 3. Web/day35 project_Account/js/MinusAccount.js
/**
 * 생성자 생성 (Account 상속)
 */
function MinusAccount(accountNum, accountOwner, passwd, restMoney, borrowMoney){
	Account.call(this, accountNum, accountOwner, passwd, restMoney);
	this.borrowMoney = borrowMoney;
}

MinusAccount.prototype = new Account('', '', 0, 0);

MinusAccount.prototype.getBorrowMoney = function(){
	return this.borrowMoney;
}
MinusAccount.prototype.setBorrowMoney = function(borrowMoney){
	this.borrowMoney = borrowMoney;
}

/**
 * @Override
 */
MinusAccount.prototype.deposit = function(money){
	Account.prototype.setRestMoney(Account.prototype.getRestMoney.call(this) + money);
	return Account.prototype.getRestMoney.call(this);
}
/**
 * @Override
 * 남은금액 return
 */
MinusAccount.prototype.getRestMoney = function(money){
	return Account.prototype.getRestMoney.call(this) - this.borrowMoney;
}

MinusAccount.prototype.toString = function(){
	var output = '';
	for ( var key in this) {
		if(key != 'toString'){
			output += key + ' = ' + this[key]+"\t";
		}
	}
	return output;
}

delete MinusAccount.prototype.accountNum;
delete MinusAccount.prototype.accountOwner;
delete MinusAccount.prototype.passwd;
delete MinusAccount.prototype.restMoney;

MinusAccount.prototype.constructor = MinusAccount;
